Gemfile

```diff
@@ -5,7 +5,8 @@ git_source(:github) {|repo_name| "https://github.com/#{repo_name}" }
 # Specify your gem's dependencies in promoted-ruby-client.gemspec
 gemspec
 
-gem 'faraday', '~> 1.4.1'
+gem 'async-http'
```

Comment:
I sent some direct chats about this.  Yea, I think eventually having a plugin for this makes sense.